lib/spack/spack/test/cmd/compiler.py

```diff
@@ -285,26 +284,7 @@ def test_compiler_list_empty(no_compilers_yaml, working_env, compilers_dir):
 \tmodules  = ['gcc/7.7.7', 'foobar']
 \toperating system  = foobar
 """,
-        ),
-        (
-            {
-                "spec": "gcc@7.7.7",
-                "prefix": "{prefix}",
-                "modules": ["gcc/7.7.7", "foobar"],
-                "extra_attributes": {"flags": {"fflags": "-ffree-form"}},
-            },
-            """gcc@7.7.7:
-\tpaths:
-\t\tcc = {compilers_dir}{sep}gcc-8{suffix}
-\t\tcxx = {compilers_dir}{sep}g++-8{suffix}
-\t\tf77 = {compilers_dir}{sep}gfortran-8{suffix}
-\t\tfc = {compilers_dir}{sep}gfortran-8{suffix}
```

Comment:
I deleted this one, cause detecting `gcc-8` for a `gcc@7.7.7` spec seems wrong. I think it's better to warn/leave a debug message and avoid auto-detection.

Reply:
See https://github.com/spack/spack/pull/43464/commits/805f78769694f4fddf464b97dad838b545d92ba7